xyz-psql-connector/src/main/java/com/here/xyz/psql/query/IterateFeatures.java

```diff
@@ -112,7 +114,7 @@ protected SQLQuery buildQuery(IterateFeaturesEvent event) throws SQLException {
     }
     else {
       if (hasHandle)
-        query.setQueryFragment("filterWhereClause", query.getQueryFragment("filterWhereClause") + " AND i > #{startOffset}");
```

Comment:
Exactly, because if _hasSearch_ is false then the search query was only _"TRUE"_. In that case the _filterWhereClause_ does not have to be extended using "AND", but it can be simply replaced completely.